handler.js

```diff
@@ -111,6 +115,19 @@ module.exports.start_build = (event, context, callback) => {
         console.log("Using an external buildspec. Will not pass sourceVersion when starting the CodeBuild job. Set CB_EXTERNAL_BUILDSPEC=false to change.");
       }
 
+      //Adding extra env variables to the CI build jobs
+      var extraEnv = getKeyValuePairs(cbEnv);
+      for(var key in extraEnv){
+        //only add env that has a name
+        if(key){
```

Comment:
this feels kinda weird... why would the key be empty/null?

Reply:
trying to catch the case where `CB_ENV="Foo=Bar;=123".` You are right. If that is the case, the user would need to fix it by themselves.

Reply:
ahh yes, let user error fail so they can handle it

Reply:
I need to switch up my mentality. Always try to catch/handle user errors when developing web applications. We don't know how creative those users can get.

Reply:
it's cool if doing it to provide a helpful log/error message back to the user, not so cool to just swallow/ignore the error condition...